sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -750,7 +750,8 @@ public AnalyzedRelation visitValues(Values values, StatementAnalysisContext cont
                 for (int i = 0; i < firstRow.size(); i++) {
                     var typeOfFirstRowAtPos = firstRow.get(i).valueType();
                     var typeOfCurrentRowAtPos = columns.get(i).valueType();
-                    if (!typeOfCurrentRowAtPos.equals(typeOfFirstRowAtPos)) {
+                    if (!typeOfFirstRowAtPos.isConvertableTo(typeOfCurrentRowAtPos)
```

Comment:
Which type will this end up using? Couldn't this lead to ClassCastExceptions due to a mixture of types?

In the other branch we've type precedence logic integrated.